build/tasks/test-on-node.js

```diff
@@ -66,31 +70,39 @@ module.exports = function( grunt ) {
 			testActive = true;
 		} );
 		QUnit.log( function( details ) {
-			if ( !testActive || details.result ) {
+			if ( !testActive || details.result || details.todo ) {
 				return;
 			}
 			var message = "name: " + details.name + " module: " + details.module +
 				" message: " + details.message;
 			grunt.log.error( message );
 		} );
-		QUnit.testDone( function() {
+		QUnit.testDone( function( details ) {
 			testActive = false;
+
+			if ( details.failed && !details.todo ) {
+				stats.failedTests++;
```

Comment:
This will fail to increment `failedTests` when a TODO test has no failing assertions. I think it should instead use the same logic as the HTML reporter.